### DISCUSSION TITLE
Help on AWS Cost Estimation and Resources Size

### DISCUSSION THREAD

**Discussion by MUHAMMAD0KASHIF** [milvus-io/milvus#24809] [Q&A and General discussion]
Hi,
I am working on cost estimation for an assignment. We have around 561 millions vectors and currently for POC purpose we are running milvus docker on aws EC2. I want to get know how much compute power would be required if we want to sotre around 561 million vectors and other details as given in attached image. Will appreciate any help. Also is it feasible to run docker on aws Ec2 for such a large amount of data.

![datasize](https://github.com/milvus-io/milvus/assets/24218558/3cb22413-53dc-41cc-9b67-bdbac50cd5b9)

**Comment by xiaofan-luan** [comment]
The QPS highly depend on your QPS and vector dimension. Could you offer more details on your scenario?

**Reply by MUHAMMAD0KASHIF** [reply]
It is all text data, and at start we're expecting 3000 queries per second at max, and vector dimension is 1536.

**Reply by xiaofan-luan** [reply]
> The QPS highly depend on your QPS and vector dimension. Could you offer more details on your scenario?

USE HNSW, I would recommend you to start from 256core 1TB memory see it meets your performance requirements
